c2rust-analyze/src/dataflow/type_check.rs

```diff
@@ -104,6 +100,7 @@ impl<'tcx> TypeChecker<'tcx, '_> {
     fn visit_cast(&mut self, cast_kind: CastKind, op: &Operand<'tcx>, to_lty: LTy<'tcx>) {
         let to_ty = to_lty.ty;
         let from_lty = self.acx.type_of(op);
+        let from_ty = from_lty.ty;
```

Comment:
nit: just pass `from_lty.ty` to `is_transmutable_ptr_cast`

Reply:
It lets me use `{from_ty:?}`, as only variables work in that position, not expressions.